src/uu/tail/src/args.rs

```diff
@@ -242,6 +237,82 @@ impl Settings {
 
         Ok(settings)
     }
+
+    pub fn has_only_stdin(&self) -> bool {
+        self.inputs.iter().all(|input| input.is_stdin())
+    }
+
+    pub fn has_stdin(&self) -> bool {
+        self.inputs.iter().any(|input| input.is_stdin())
+    }
+
+    pub fn num_inputs(&self) -> usize {
+        self.inputs.len()
+    }
+
+    pub fn check_warnings(&self) -> CheckResult {
+        if self.retry {
+            if self.follow.is_none() {
+                show_warning!("--retry ignored; --retry is useful only when following");
+            } else if self.follow == Some(FollowMode::Descriptor) {
+                show_warning!("--retry only effective for the initial open");
+            }
+        }
+
+        let mut no_pid_support: Option<CheckResult> = None;
+        if self.pid != 0 {
+            if self.follow.is_none() {
+                show_warning!("PID ignored; --pid=PID is useful only when following");
+            } else if !platform::supports_pid_checks(self.pid) {
+                show_warning!("--pid=PID is not supported on this system");
+                no_pid_support = Some(CheckResult::NoPidSupport);
+            }
+        }
+
+        if self.follow.is_some() && !self.default_stdin && self.has_stdin() {
```

Comment:
What's the test case for this warning? I can't think of any reason why it should only apply with an explicitly given stdin (i.e `!self.default_stdin`)

Reply:
This warnings should only be printed when you explicitely follow a redirected tty (among some special circumstances) to warn about the intention as being inefficient. Maybe also a look into gnu's implementation of tail regarding this warning explains this a little bit more.

Reply:
> Maybe also a look into gnu's implementation of tail

I don't want to have to do that :) And I'm not speaking as a reviewer here, I'm fine with checking against GNU, but I think it should be obvious from the code why this is happening, so please add a similar explanation here.

Reply:
I haven't documented anything, right now. I'm going to do this.

Reply:
done

Reply:
So I looked into the GNU code and I think the `default_stdin` is wrong. The equivalent in their code is `found_hyphen` which is determined from the list of files. But if no files are given then this list is set to `"-"`, so `found_hyphen` will be true for when no `default_stdin` is given. I think `found_hyphen` is just exactly `self.has_stdin()`.

Reply:
You're right, `self.has_stdin()` is the exact equivalent of `found_hyphen`. I adjusted the if clause to match gnu's tail more closely.